core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -96,6 +96,10 @@ public void initialize(String name, Map<String, String> props) {
     this.io = CatalogUtil.loadFileIO(ioImpl != null ? ioImpl : ResolvingFileIO.class.getName(), properties, conf);
   }
 
+  protected Map<String, String> headers() {
+    return ImmutableMap.of();
```

Comment:
Right now, headers aren't used other than overriding this in tests to validate that they can be set and are passed through the catalog correctly.